### ISSUE TITLE
Unnecessary parens after 'return' keyword issued when returning a generator expression

### ISSUE THREAD

**Issue by DervishD** [PyCQA/pylint#5824]
### Bug description

`pylint` produces `superfluous-parens` error when a generator expression is returned and that generator expression contains additional parentheses, at least that's what I infer since a simple generator expression like `(i for i in ['a', 'b', 'c'])` does not trigger the error.

Below is the minimum code I've been able to create that reproduces the problem. Empty docstrings are needed to supress common warnings about missing docstrings, etc. Other than that they don't seem important.

```
'''.'''


def test_fun():
    '''.'''
    return ((i**i, value.upper()) for i, value in enumerate(['a', 'b', 'c']))

```
 

### Configuration

```ini
Default configuration
```

### Command used

```shell
pylint test.py

and also

pylint --rcfile=NUL
```

### Pylint output

```shell
************* Module test
test.py:6:0: C0325: Unnecessary parens after 'return' keyword (superfluous-parens)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

Well, that error should not be raised when returning generator expressions as the outer parentheses are in fact needed. This is a false positive, IMHO.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.2
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10 x64 here, latest stable build (19044.1526) for 21H2.

**Comment by mbyrnepr2** [comment]
I can reproduce it on the 2.10.2 version but it works ok for me on 2.12.0 @DervishD if you could try that.

pylint version:
```
pylint 2.12.0
astroid 2.9.3
Python 3.10.0b2
```

**Comment by DervishD** [comment]
@mbyrnepr2 you're right, it does **NOT** happen in `pylint 2.12.0`.

For some reason, `pip list --outdated` was not showing a new version for `pylint` when I was trying to update it before reporting the problem. Also, I did not find anything in closed bugs about this particular issue. The first part is somehow `pip`'s fault, but the second part is entirely my fault, I obviously did not search properly.

Thanks a lot!

**Comment by mbyrnepr2** [comment]
@DervishD no worries - thank you for taking the care to highlight a potential issue!

**Comment by DervishD** [comment]
My pleasure, really, and thanks a lot for `pylint`, of course 😃.

**Comment by mbyrnepr2** [comment]
I'm going to pass this forward: Pylint is awesome thanks to @Pierre-Sassoulas & a multitude of others; what I do is a drop in the ocean compared to these folks 🚀

**Comment by DervishD** [comment]
Of course! 💪🏼 😃
